### PR TITLE
Filter empty rows when looking for the header in `findHeader`

### LINKED ISSUE #16: `findHeader` finds a wrong header
For https://gitlab.com/bath_open_instrumentation_group/sangaboard/-/blob/master/pcb_design/v0.3/Sangaboard%20v3%20BOM.xlsx the row with the most columns isn't the header.

It returns line 30 as the header

image

##
https://github.com/kitspace/npm-1-click-bom/blob/cf882824fd1b9f825802195c32206d9cb559bd65/src/parser.js#L330-L344

### PR DESCRIPTION
**Pull request by AbdulrhmnGhanem**
Fixes #16

### CODE REVIEW

**Comment by kasbah** [comment]
I thought I left a comment on this before. This could be simplified by filtering out empty cells before counting them.

**Review by kasbah — CHANGES_REQUESTED** [review]
This can be simplified by filtering out empty cells before counting them.

File: src/parser.js [hunk 1]
```diff
@@ -327,11 +327,11 @@ function processLine(warnings, line, i) {
   return newLine
 }
 
-// finds the first row with the most columns
+// finds the first row with the most columns and least empty cells
 function findHeader(aoa) {
   const {i} = aoa.reduce(
     (prev, row, i) => {
-      const len = row.length
+      const len = row.filter(cell => cell != null).length
```
Comment (kasbah) [inline src/parser.js:334]: Are they `null`? Not `''`?

Comment (AbdulrhmnGhanem) [reply src/parser.js:334]: Yes. it's a cell object, not the cell content.

**Pull request merged**